app/tournament/tournament.js

```diff
@@ -0,0 +1,93 @@
+import {Player, validatePlayer} from "./player";
+import {Match, ChildMatch} from "./match";
+import * as utils from "../common/utils";
+
+export class Tournament {
+    constructor() {
+        this.players = [];
+        this.matches = [];
+    }
+
+    addPlayer(player) {
+        this.players.push(new Player(player.name, player.rank));
+        sortPlayersByRank(this);
+    }
+
+    addMatch(match) {
+        this.matches.push(match);
+    }
+
+    getRounds() {
+        return this.matches[this.matches.length - 1].round;
+    }
+
+    generate() {
+        generateTournamentBracket(this);
+    }
+
+    validatePlayer(player) {
+        var validationResult = validatePlayer(player);
+
+        if (validationResult.success && this.existsPlayer(player)) {
+            validationResult.error = "Player already enrolled";
+            validationResult.success = false;
+        }
+
+        return validationResult;
+    }
+
+    existsPlayer(player) {
+        return this.players.find( function(p) {
+            return p.name.toLowerCase() === this.name.toLowerCase()
+            || p.rank === this.rank;
+        }, player);
+    }
+}
+
+function sortPlayersByRank(tournament) {
+    tournament.players.sort( function(p1, p2) {
+        return p1.rank - p2.rank;
+    });
+}
+
+function generateTournamentBracket(tournament) {
```

Comment:
I made that method in an attempt to keep the class body short and concise. It has no value in terms of implementation but looks more organised to me that way.